log4j-api/src/main/java/org/apache/logging/log4j/util/Strings.java

```diff
@@ -288,6 +289,19 @@ public static String trimToNull(final String str) {
         return isEmpty(ts) ? null : ts;
     }
 
+    /**
+     * Removes control characters from both ends of this String returning {@code Optional.empty()} if the String is
+     * empty ("") after the trim or if it is {@code null}.
+     * @param str The String to trim.
+     * @return An Optional containing the String.
+     *
+     * @see #trimToNull(String)
+     * @since 2.24.0
+     */
+    public static Optional<String> trimToOptional(final String str) {
```

Comment:
Where do we use this?

Reply:
In about 10 locations in `log4j-plugins`, but I can move it there.